datajoint/expression.py

```diff
@@ -119,17 +121,33 @@ def where_clause(self):
             else " WHERE (%s)" % ")AND(".join(str(s) for s in self.restriction)
         )
 
+    def sorting_clauses(self):
+        if not self._top:
+            return ""
+        clause = (
+            ""
+            if not any(_flatten_attribute_list(self.primary_key, self._top.order_by))
+            else f" ORDER BY \
+                {', '.join(_flatten_attribute_list(self.primary_key, self._top.order_by))}"
+        )
```

Comment:
we can avoid `calling _flatten_attribute_list` twice:
```suggestion
        clause = ', '.join(_flatten_attribute_list(self.primary_key, self._top.order_by)) 
        if clause:
            clause = f" ORDER BY {clause}"
```